app/actions/websocket/channel.ts

```diff
@@ -18,42 +27,201 @@ import {getCurrentUser, getUserById} from '@queries/servers/user';
 import {dismissAllModals, popToRoot} from '@screens/navigation';
 import {isTablet} from '@utils/helpers';
 
-export async function handleUserAddedToChannelEvent(serverUrl: string, msg: any) {
-    const database = DatabaseManager.serverDatabases[serverUrl];
-    if (!database) {
+// Received when current user created a channel in a different client
+export async function handleChannelCreatedEvent(serverUrl: string, msg: any) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
         return;
     }
-    const currentUser = await getCurrentUser(database.database);
-    const {team_id: teamId, user_id: userId} = msg.data;
-    const {channel_id: channelId} = msg.broadcast;
-    const models: Model[] = [];
+
+    const {team_id: teamId, channel_id: channelId} = msg.data;
 
     try {
-        const addedUser = getUserById(database.database, userId);
-        if (!addedUser) {
-            // TODO Potential improvement https://mattermost.atlassian.net/browse/MM-40581
-            const {users} = await fetchUsersByIds(serverUrl, [userId], true);
-            if (users) {
-                models.push(...await database.operator.handleUsers({users, prepareRecordsOnly: true}));
+        const models: Model[] = [];
+        const {channels, memberships} = await fetchMyChannel(serverUrl, teamId, channelId, true);
+        if (channels && memberships) {
+            const prepare = await prepareMyChannelsForTeam(operator, teamId, channels, memberships);
+            if (prepare) {
+                const prepareModels = await Promise.all(prepare);
+                const flattenedModels = prepareModels.flat();
+                if (flattenedModels?.length > 0) {
+                    models.push(...flattenedModels);
+                }
+                const categoryModels = await addChannelToDefaultCategory(serverUrl, channels[0], true);
+                if (categoryModels.models?.length) {
+                    models.push(...categoryModels.models);
+                }
             }
         }
+        operator.batchRecords(models);
+    } catch {
+        // do nothing
+    }
+}
+
+export async function handleChannelUnarchiveEvent(serverUrl: string, msg: any) {
+    try {
+        await setChannelDeleteAt(serverUrl, msg.data.channel_id, 0);
```

Comment:
not really, in my mind this can also be about a channel that was archived but we are not a member of, but the function does check for the channel to exist in the DB.

That said, you need to specify a `deleteAt` argument, `Date.now` would suffice.

```suggestion
        await setChannelDeleteAt(serverUrl, msg.data.channel_id, Date.now());
```

Reply:
This is to "unarchive" (reopen) a channel, not to archive (close) a channel. Therefore, the deleteAt goes to 0.